statsd/telemetry_test.go

```diff
@@ -163,20 +252,30 @@ func TestTelemetryWithAggregationDevMode(t *testing.T) {
 
 	telemetry := newTelemetryClient(client, "test_transport", true)
 
-	expectedMetrics := map[string]int64{
-		"datadog.dogstatsd.client.aggregated_context": 5,
-	}
-	for k, v := range basicExpectedMetrics {
-		expectedMetrics[k] = v
-	}
-	for k, v := range devModeExpectedMetrics {
-		expectedMetrics[k] = v
-	}
-	for k, v := range devModeAggregationExpectedMetrics {
-		expectedMetrics[k] = v
-	}
+	expectedMetrics := []metric{}
+	expectedMetrics = appendBasicMetrics(expectedMetrics, nil)
+	expectedMetrics = appendAggregationMetrics(expectedMetrics, nil, true)
+	expectedMetrics = appendDevModeMetrics(expectedMetrics, nil)
+
+	testTelemetry(t, telemetry, expectedMetrics)
+}
+
+func TestTelemetryWithAggregationDevModeWithGlobalTags(t *testing.T) {
+	os.Setenv("DD_ENV", "test")
+	defer os.Unsetenv("DD_ENV")
```

Comment:
Note: If `DD_ENV` was set, this test unset `DD_ENV`

Reply:
+1